src/Package.zig

```diff
@@ -608,6 +608,20 @@ fn fetchAndUnpack(
             // I have not checked what buffer sizes the xz decompression implementation uses
             // by default, so the same logic applies for buffering the reader as for gzip.
             try unpackTarball(gpa, prog_reader.reader(), tmp_directory.handle, std.compress.xz);
+
+            tmp_directory = d: {
```

Comment:
Why only reopen in the `application/x-xz` case? Wouldn't this affect all the other cases too?